packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+module.exports = (filePath, options, jasmine, snapshotState) => ({
+  toMatchSnapshot: (util, customEquality) => {
+    return {
+      compare(rendered, expected) {
+
+        const specRunningFullName = snapshotState.specRunningFullName;
+
+        if (expected !== undefined) {
+          throw new Error('toMatchSnapshot() does not accepts parameters.');
+        }
+        if (typeof rendered !== 'string') {
+          throw new Error('toMatchSnapshot() only works with Strings.');
+        }
+
+        const snapshot = snapshotState.snapshot;
+
+        const specRunningCallCounter = (
+            snapshotState.specsNextCallCounter[specRunningFullName]
+        );
+
+        let pass = false;
+        let message;
+        let res = {};
+        const key = specRunningFullName + ' ' + specRunningCallCounter;
+
+        if (!snapshot.fileExists()) {
```

Comment:
if by empty you mean '{}' it shouldn't exists, because it's never created by jest-snapshot, 
but as you also say in that case it will work correctly because it will miss the property check, so it will add the missing snapshots (I was doing this myself before adding the -u flag)